RELEASE.md

```diff
@@ -0,0 +1,20 @@
+# Release Process
+
+The process to release a new version of cri-resource-manager is:
+
+- [ ] File [a new issue](https://github.com/intel/cri-resource-manager/issues/new)
+  to propose a new release. Copy this checklist into the issue description.
+- [ ] In the issue description, add a changelog section, describing changes
+  since the last release.
+- [ ] Run `git tag -a -m "CRI Resource Manager release $VERSION" $VERSION`.
```

Comment:
I would suggest adding
```
 - [ ] run `make release-tests` to run an extended set of tests prior to a release
```
And add already now an empty `release-tests` target to our `Makefile` which we then start
populating with things like running end-to-end test, etc. as we go/get them up and running.

Reply:
Should we add steps covering our internally mandated other checks prior to a release ?

Reply:
Yeah, I'll add these 👍

Reply:
Added both